### DISCUSSION TITLE
TVheadend

### DISCUSSION THREAD

**Discussion by blaumsass** [tteck/Proxmox#570] [Ideas]
A LXC container witch tvheadend would i find be good.

**Comment by RealJustMe** [comment]
+1 would love to have a script for this

**Comment by Vakrehus** [comment]
+1 TVheadend would be nice.

**Comment by diogopms** [comment]
I agree 👍🏻 TVheadend

**Comment by flat0utty** [comment]
Would love to see this 👍

**Comment by ketzerkadaver** [comment]
+1 You would make my day!

**Comment by KinDR007** [comment]
+1

**Comment by iSp33dy** [comment]
+1

**Comment by naijaping** [comment]
+1

**Comment by tteck** [comment]
Create a Debian 12 LXC
```
bash -c "$(wget -qLO - https://dl.cloudsmith.io/public/tvheadend/tvheadend/setup.deb.sh)"
apt update  
apt install tvheadend
```
IP:9981

**Reply by iSp33dy** [reply]
Which Version?

**Reply by tteck** [reply]
HTS Tvheadend 4.3-2098~g44bf691ac

**Reply by iSp33dy** [reply]
Great... thanks for this... i will Update my tvHeadend on debian 11 to this Version.. 🤙

**Comment by manuelkamp** [comment]
how did you all got your tv card working in LXC? I have a VM with tvheadend and migrated an LXC from there. VM works for years now, but cannot get LXC to work because TV card is not showing up in LXC. It is connected via USB, it is a Hauppauge dualHD. Tried almost all instructions on proxmox forum and google top results...?

**Reply by iSp33dy** [reply]
enable the experimental device nodes... this enables the passthrough from your usb, and add the device group to the lxc conf file like this:

`
lxc.cgroup2.devices.allow: c *:* rwm
`
`
lxc.mount.entry: /dev/dvb dev/dvb none bind,optional,create=dir
`

have fun..

**Reply by manuelkamp** [reply]
thanks, but that is what I meant by following almost all instructions and google top results. This (only this alone) does not work for me. There is more to do. I can see the device with lsusb in the LXC but it is not shown in tvheadend. So I found that changing rights of /dev/dvb etc should do the trick, but also this did not help me. So i figured out, that it must be somewhat specific and tried my luck here. It seems it just has to stay in a VM on my side...

**Reply by iSp33dy** [reply]
But the TVH VM is switched off when you try to set up the LXC? otherwise no free adapter will be found. Rights and users must match the path to TVH. then it should work too...

**Reply by sherbibv** [reply]
I also have a problem with this approach. I successfully passed the USB TV tuner to the tvheadend lxc but if I run powertop --auto-tune I get 99% of the time continuity errors. If I disable the optimization for the USB port where the tuner is, the errors dissappear. But if I leave the optimization off I cant get a higher c state (is stuck at c2 vs c6 with the optimization on). What can I do to have the optimization on and still have no continuity errors?

**Reply by manuelkamp** [reply]
I think that my problem has to do with the fact, that I had a hauppauge quad pci-e card previously installed (2 years ago). I had to remove it and use this dual USB card, because I needed the pci lanes for 10gpbs lan card. I see, that 4 dvb devices get passed through, altough the usb only has 2? how to fix that?
